run_project_tests.py

```diff
@@ -603,6 +604,14 @@ def force_regenerate():
 
     return testresult
 
+def set_boost_nativefile_paths(test: TestDef) -> None:
+    script_dir = os.path.abspath(os.path.split(__file__)[0])
+
+    confdata = ConfigurationData()
+    confdata.values = {'ROOT': (script_dir, 'project base directory')}
+
+    mesonlib.do_conf_file(str(test.path / 'nativefile.ini.in'), str(test.path / 'nativefile.ini'), confdata, 'meson')
```

Comment:
Please put the generated files into the build dir and not into the project dir. Polluting the project dir can still be a pain for running tests locally, even if they are in the .gitignore.

You will have access to the build dir if you move the function call to here: https://github.com/mesonbuild/meson/blob/master/run_project_tests.py#L498. Maybe you could make a generic `detect_cross_args() -> Path, Path` function that handles the configuration and detection of any cross/native files.